src/Files.App/Helpers/GitHelpers.cs

```diff
@@ -266,6 +268,104 @@ public static async void PullOrigin(string? repositoryPath)
 			IsExecutingGitAction = false;
 		}
 
+		public static bool IsRepositoryEx(string path, out string repoRootPath)
+		{
+			repoRootPath = path;
+
+			var rootPath = SystemIO.Path.GetPathRoot(path);
+			if (rootPath is null)
+				return false;
+
+			var repositoryRootPath = GetGitRepositoryPath(path, rootPath);
+			if (string.IsNullOrEmpty(repositoryRootPath))
+				return false;
+
+			if (Repository.IsValid(repositoryRootPath))
+			{
+				repoRootPath = repositoryRootPath;
+				return true;
+			}
+
+			return false;
+		}
+
+		public static GitItemModel GetGitInformationForItem(Repository repository, string path)
+		{
+			var rootRepoPath = repository.Info.WorkingDirectory;
+			var relativePath = path.Substring(rootRepoPath.Length).Replace('\\', '/');
+
+			var commit = GetLastCommitForFile(repository, relativePath);
+			//var commit = repository.Commits.QueryBy(relativePath).FirstOrDefault()?.Commit; // Considers renames but slow
+
+			var changeKind = ChangeKind.Unmodified;
+			//foreach (TreeEntryChanges c in repository.Diff.Compare<TreeChanges>())
+			foreach (TreeEntryChanges c in repository.Diff.Compare<TreeChanges>(repository.Commits.FirstOrDefault()?.Tree, DiffTargets.Index | DiffTargets.WorkingDirectory))
+			{
+				if (c.Path.StartsWith(relativePath))
+				{
+					changeKind = c.Status;
+					break;
+				}
+			}
+
+			string? changeKindHumanized = "";
```

Comment:
I don't think this one should be nullable, I mean you always assign a value to it

Reply:
let me fix after 2.5